File: Unity Pepijn/Melee/Assets/MeleeSystem.js
#pragma strict

var Damage : int = 50;
var Distance : float;

function Update ()
{
	if (Input.GetButtonDown("Fire1"))
	{
		var hit : RaycastHit;
		if (Physics.Raycast (transform.position, transform.TransformDirection(Vector3.forward), hit))
		{
			Distance = hit.distance;
			hit.transform.SendMessage("ApplyDamage", Damage, SendMessageOptions.DontRequireReceiver);
		}
	}
}

File: Unity Pepijn/Melee/Assets/MeleeSystem.cs
using UnityEngine;
using System.Collections;

public class MeleeSystem : MonoBehaviour {
	
	
	int Damage = 50;
	float Distance;
	
	void  Update (){
		if (Input.GetButtonDown("Fire1"))
		{
			RaycastHit hit;
			if (Physics.Raycast (transform.position, -Vector3.up, out hit))
			{
				Distance = hit.distance;
				hit.transform.SendMessage("ApplyDamage", Damage, SendMessageOptions.DontRequireReceiver);
			}
		}
	}
}